src/include/duckdb/execution/operator/persistent/csv_scanner/csv_state_machine_cache.hpp

```diff
@@ -0,0 +1,30 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/execution/operator/persistent/csv_scanner/csv_state_machine_cache.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "duckdb/execution/operator/persistent/csv_scanner/csv_reader_options.hpp"
+#include "duckdb/execution/operator/persistent/csv_scanner/csv_buffer_manager.hpp"
+
+namespace duckdb {
+typedef uint8_t state_machine_t[7][256];
```

Comment:
It's a bit confusing to have the type definition of the state machine outside of the `csv_state_machine.hpp`. And I think we want to make `7` (and potentially `256`) `static constexpr`, so we can change them in a single place (probably `CSVStateMachine`) and don't have 'magic' numbers around.